File: src/components/pages/projects/$projectId/shared/connect-snippets/android/java/AppwriteClient.java
import android.content.Context;

import io.appwrite.Client;

public final class AppwriteClient {
  private static volatile Client instance;

  private AppwriteClient() {}

  public static Client get(Context context) {
    if (instance == null) {
      synchronized (AppwriteClient.class) {
        if (instance == null) {
          String endpoint = BuildConfig.APPWRITE_ENDPOINT;
          instance = new Client(
                  context.getApplicationContext(),
                  endpoint,
                  endpoint.replaceFirst("http", "ws"))
              .setProject(BuildConfig.APPWRITE_PROJECT_ID);
        }
      }
    }
    return instance;
  }
}

File: src/components/pages/projects/$projectId/shared/connect-snippets/android/kotlin/AppwriteClient.kt
import android.content.Context
import io.appwrite.Client

object AppwriteClient {
  @Volatile private var instance: Client? = null

  fun get(context: Context): Client =
    instance ?: synchronized(this) {
      instance ?: Client(context.applicationContext)
        .setEndpoint(BuildConfig.APPWRITE_ENDPOINT)
        .setProject(BuildConfig.APPWRITE_PROJECT_ID)
        .also { instance = it }
    }
}
